segfault on certificate revocation when rate limit is hit

### Welcome

- [x] Yes, I'm using a binary release within the two latest releases.
- [x] Yes, I've searched for similar issues on GitHub and didn't find any.
- [x] Yes, I've included all information below (version, config, etc).

### What did you expect to see?

While experimenting on LetsEncrypt's stage environment, during a next attempt to revoke a certificate (after several requests/revocations) I got segfault.

Since it didn't happen previously, and after some time (and repeated attempts, which also produced same segfault) I was able to proceed, I _believe_ that this happens when rate limit at LetsEncrypt is hit.


### What did you see instead?

A segfault (see details for stack trace).

### How do you use lego?

Binary

### Reproduction steps

1. Create many certificates at LetsEncrypt's stage environment (I requested 10)
2. Revoke all of them
3. Repeat from step 1 few times in a row - eventually it will segfault (at least I was able to reproduce it this way)

### Effective version of lego

```console
lego version 4.28.0 linux/amd64
```

### Logs

<details>

```console
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x20f4634]

goroutine 1 [running]:
github.com/go-acme/lego/v4/cmd.checkRetry({0x2e7ea40?, 0x41aa180?}, 0x0, {0x2e535a0?, 0xc00047ede0?})
	github.com/go-acme/lego/v4/cmd/setup.go:183 +0x54
github.com/hashicorp/go-retryablehttp.(*Client).Do(0xc000402080, 0xc00044c2b8)
	github.com/hashicorp/go-retryablehttp@v0.7.8/client.go:722 +0x2f5
github.com/hashicorp/go-retryablehttp.(*RoundTripper).RoundTrip(0xc00044c288, 0xc000446500?)
	github.com/hashicorp/go-retryablehttp@v0.7.8/roundtripper.go:47 +0x6a
github.com/go-acme/lego/v4/acme/api/internal/sender.(*httpsOnly).RoundTrip(0xc000680008?, 0x2e53820?)
	github.com/go-acme/lego/v4/acme/api/internal/sender/sender.go:180 +0x82
net/http.send(0xc000446500, {0x2e53820, 0xc00071d5c0}, {0x1?, 0x5e7526?, 0x0?})
	net/http/client.go:259 +0x5e2
net/http.(*Client).send(0xc000724c90, 0xc000446500, {0x0?, 0xc000611068?, 0x0?})
	net/http/client.go:180 +0x91
net/http.(*Client).do(0xc000724c90, 0xc000446500)
	net/http/client.go:729 +0x9c9
net/http.(*Client).Do(...)
	net/http/client.go:587
github.com/go-acme/lego/v4/acme/api/internal/sender.(*Doer).do(0x0?, 0xc000446500, {0x22cb5e0, 0xc000451a40})
	github.com/go-acme/lego/v4/acme/api/internal/sender/sender.go:90 +0x58
github.com/go-acme/lego/v4/acme/api/internal/sender.(*Doer).Get(0xc00044c2a0, {0x7fffaba060d2?, 0xc000611390?}, {0x22cb5e0, 0xc000451a40})
	github.com/go-acme/lego/v4/acme/api/internal/sender/sender.go:46 +0x77
github.com/go-acme/lego/v4/acme/api.getDirectory(0xc00044c2a0, {0x7fffaba060d2, 0x36})
	github.com/go-acme/lego/v4/acme/api/api.go:151 +0x8b
github.com/go-acme/lego/v4/acme/api.New(0xc000724c90, {0xc000727fa1, 0xf}, {0x7fffaba060d2, 0x36}, {0xc0003aa140, 0x40}, {0x288ec00, 0xc0007245d0})
	github.com/go-acme/lego/v4/acme/api/api.go:41 +0x1a8
github.com/go-acme/lego/v4/lego.NewClient(0xc000611860)
	github.com/go-acme/lego/v4/lego/client.go:48 +0xe5
github.com/go-acme/lego/v4/cmd.newClient(0xc00048e880, {0x2e7bc80, 0xc000724600}, {0x2a2dc58, 0x4})
	github.com/go-acme/lego/v4/cmd/setup.go:86 +0x3f5
github.com/go-acme/lego/v4/cmd.revoke(0xc00048e880)
	github.com/go-acme/lego/v4/cmd/cmd_revoke.go:47 +0x10d
github.com/urfave/cli/v2.(*Command).Run(0xc0001f4c60, 0xc00048e880, {0xc00071d060, 0x1, 0x1})
	github.com/urfave/cli/v2@v2.27.7/command.go:276 +0x7c2
github.com/urfave/cli/v2.(*Command).Run(0xc0001f5340, 0xc00043dac0, {0xc000050070, 0x7, 0x7})
	github.com/urfave/cli/v2@v2.27.7/command.go:269 +0xa30
github.com/urfave/cli/v2.(*App).RunContext(0xc000219200, {0x2e7ea40, 0x41aa180}, {0xc000050070, 0x7, 0x7})
	github.com/urfave/cli/v2@v2.27.7/app.go:333 +0x5a5
github.com/urfave/cli/v2.(*App).Run(...)
	github.com/urfave/cli/v2@v2.27.7/app.go:307
main.main()
	github.com/go-acme/lego/v4/cmd/lego/main.go:41 +0x1df

```

</details>

### Go environment (if applicable)

_No response_

--- a/cmd/setup.go
+++ b/cmd/setup.go
@@ -180,6 +180,10 @@ func checkRetry(ctx context.Context, resp *http.Response, err error) (bool, erro
 		return rt, err
 	}
 
+	if resp == nil {
+		return rt, nil
+	}
+
 	if resp.StatusCode/100 == 2 {
 		return rt, nil
 	}

